lib/adapters/class-vip-enterprise-search.php

```diff
@@ -20,141 +17,112 @@
+
+		// Determine whether we should filter the list or not.
+		$restricted_post_types = $this->get_restricted_post_types();
+		if ( empty( $restricted_post_types ) ) {
+			return $post_types;
 		}
 
-		// TODO Write DSL for date faceting as configured.
+		// Rebuild the list of post types using the allowlist.
+		$filtered_post_types = [];
+		foreach ( $restricted_post_types as $post_type ) {
+			$filtered_post_types[ $post_type ] = $post_type;
+		}
 
-		return $query;
+		return $filtered_post_types;
 	}
 
 	/**
-	 * Allow empty searching in conjunction with faceting.
-	 * Filters `ep_post_formatted_args`.
-	 *
-	 * Since EP is not "expecting"  our custom faceting,
-	 * it does a match_all when no search query string is present.
-	 * For that same reason, if there are no facets, EP's match_all is required.
+	 * A callback for the ep_post_formatted_args filter hook. Allows empty
+	 * searches, if enabled, and applies any requested aggregation filters.
 	 *
-	 * @see \ElasticPress\Indexable\Post\Post::format_args() For the `ep_post_formatted_args` filter.
+	 * @param array $formatted_args Elasticsearch query arguments.
+	 * @param array $args           WordPress query arguments.
 	 *
-	 * @param array $formatted_args Formatted ES args.
-	 * @param array $args           WP args.
+	 * @return array The modified Elasticsearch query arguments.
 	 */
-	public function allow_empty_search( $formatted_args, $args ) {
-		if (
-			isset( $args['s'] )
-			&& '' === $args['s']
-			&& ! empty( $args['fs'] )
-			&& true === $this->empty_search_faceting
-		) {
+	public function filter__ep_post_formatted_args( $formatted_args, $args ) {
+		// Enable empty search, if requested.
+		if ( $this->get_allow_empty_search() && ! empty( $args['s'] ) ) {
 			unset( $formatted_args['query']['match_all'] );
 		}
+
+		// Add requested aggregations.
+		foreach ( $this->get_aggregations() as $aggregation ) {
+			$filter = $aggregation->filter();
+			if ( ! empty( $filter ) ) {
+				$formatted_args['query']['function_score']['query']['bool']['must'][] = $filter;
+			}
+		}
+
 		return $formatted_args;
 	}
 
 	/**
-	 * Filters ElasticPress request query args to apply registered customizations.
-	 * Filters `ep_query_request_args`.
+	 * A callback for the ep_query_request_args filter. Adds aggregations to the
+	 * request so that the response will include aggregation buckets.
 	 *
-	 * @see \ElasticPress\Elasticsearch::query() For the `ep_query_request_args` filter.
+	 * @param array $request_args The request args to be filtered.
 	 *
-	 * @param array  $request_args Request arguments.
-	 * @param string $path         Request path.
-	 * @param string $index        Index name.
-	 * @param string $type         Index type.
-	 *
-	 * @return array New request arguments.
+	 * @return array The filtered request args.
 	 */
-	public function filter_ep_query_request_args( $request_args, $path, $index, $type ): array {
+	public function filter__ep_query_request_args( $request_args ): array {
```

Comment:
There's no param type hint here, but an array is assumed on the next line.  Does this need a check for an array or a type hint?

Reply:
Good idea to do the array type check. We try to avoid putting type hints on filter callbacks, because we might get something unexpected back from another filter call, but it would be better to validate the assumption at the top of the function before proceeding. I'll make this change.